Compilation error while using itk::RelabelComponentImageFilte::GetSizeOfObjectInPhysicalUnits

<!-- The text within this markup is a comment, and is intended to provide
guidelines to open an issue for the ITK repository. This text will not
be part of the issue. -->


<!-- Before submitting an issue, please check that your issue has not been
already filed. -->

### Description

Using the `itkRelabelComponentImageFilter` filter `GetSizeOfObjectInPhysicalUnits` with signed integer image causes a compilation error due comparison of integer expressions of different signedness.

### Steps to Reproduce

1. Install & build ITK (master)
2. Build [test snippet](https://gist.github.com/kenavolic/acd6a544e58a995f395812c6e404b9de)

### Expected behavior

Should be able to use the class method

### Actual behavior

Compilation error:
```shell
In file included from /home/anon/Dev/test_itk/main.cpp:2:
/home/anon/Dev/ITK/Modules/Segmentation/ConnectedComponents/include/itkRelabelComponentImageFilter.h: In instantiation of ‘float itk::RelabelComponentImageFilter<TInputImage, TOutputImage>::GetSizeOfObjectInPhysicalUnits(itk::RelabelComponentImageFilter<TInputImage, TOutputImage>::LabelType) const [with TInputImage = itk::Image<short int, 3>; TOutputImage = itk::Image<short int, 3>; itk::RelabelComponentImageFilter<TInputImage, TOutputImage>::LabelType = short int]’:
/home/anon/Dev/test_itk/main.cpp:13:46:   required from here
/home/anon/Dev/ITK/Modules/Segmentation/ConnectedComponents/include/itkRelabelComponentImageFilter.h:228:24: error: comparison of integer expressions of different signedness: ‘itk::RelabelComponentImageFilter<itk::Image<short int, 3>, itk::Image<short int, 3> >::LabelType’ {aka ‘short int’} and ‘const SizeValueType’ {aka ‘const long unsigned int’} [-Werror=sign-compare]
  228 |     if (obj > 0 && obj <= m_NumberOfObjects)
```

### Reproducibility

Always (under the same compilation flags)

### Versions

Current developpement version and older ones (detected on ITK5.2)

### Environment

* Ubuntu 18.04
* gcc 10.3
* cmake 3.25

### Additional Information

[Link to discourse discussion](https://discourse.itk.org/t/itk-getsizeofobjectinphysicalunits-signature/7237)

--- a/Modules/Segmentation/ConnectedComponents/include/itkRelabelComponentImageFilter.h
+++ b/Modules/Segmentation/ConnectedComponents/include/itkRelabelComponentImageFilter.h
@@ -209,7 +209,7 @@ public:
   ObjectSizeType
   GetSizeOfObjectInPixels(LabelType obj) const
   {
-    if (obj > 0 && obj <= m_NumberOfObjects)
+    if (obj > 0 && static_cast<SizeValueType>(obj) <= m_NumberOfObjects)
     {
       return m_SizeOfObjectsInPixels[obj - 1];
     }
@@ -225,7 +225,7 @@ public:
   float
   GetSizeOfObjectInPhysicalUnits(LabelType obj) const
   {
-    if (obj > 0 && obj <= m_NumberOfObjects)
+    if (obj > 0 && static_cast<SizeValueType>(obj) <= m_NumberOfObjects)
     {
       return m_SizeOfObjectsInPhysicalUnits[obj - 1];
     }

